File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:

        slow = head
        fast = head

        while fast != None and fast.next != None:
            fast = fast.next.next
            slow = slow.next

        
        head2 = slow
        nxt = None
        prev = None

        while head2 != None:
            nxt = head2.next
            head2.next = prev
            prev = head2
            head2 = nxt
        
        head2 = prev
        curr = head
        while(head2 != None):
            if curr.val != head2.val:
                return False
            head2 = head2.next
            curr = curr.next

        return True

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        
        ArrayList<Integer> nodes = new ArrayList<>();

        ListNode curr = head;
        while(curr != null)
        {
            nodes.add(curr.val);
            curr = curr.next;
        }

        int l=0, r=nodes.size()-1;

        while(l <= r)
        {
            if(nodes.get(l) != nodes.get(r)) return false;
            l++;
            r--;
        }

        return true;
    }
}